tsl/src/chunk.c

```diff
@@ -507,3 +508,188 @@ chunk_drop_replica(PG_FUNCTION_ARGS)
 
 	PG_RETURN_VOID();
 }
+
+static List *
+chunk_id_list_create(ArrayType *array)
+{
+	/* create a sorted list of chunk ids from array */
+	ArrayIterator it;
+	Datum id_datum;
+	List *id_list = NIL;
+	bool isnull;
+
+	it = array_create_iterator(array, 0, NULL);
+	while (array_iterate(it, &id_datum, &isnull))
+	{
+		if (isnull)
+			ereport(ERROR,
+					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					 errmsg("chunks array arguments cannot be NULL")));
+		id_list = lappend_int(id_list, DatumGetInt32(id_datum));
+	}
+	array_free_iterator(it);
+
+	(void) list_sort_compat(id_list, list_int_cmp_compat);
+	return id_list;
+}
+
+static List *
+chunk_id_list_exclusive_right_merge_join(const List *an_list, const List *dn_list)
```

Comment:
Hmm, not sure the name with `merge_join` makes sense.

Reply:
We can rename it if people like, but it is essentially what the function do